QuizApp/Search/SearchView.swift

```diff
@@ -0,0 +1,55 @@
+import SwiftUI
+import Factory
+
+struct SearchView: View {
+
+    @ObservedObject var viewModel: QuizViewModel
+
+    @State private var searchText = ""
+
+    var body: some View {
+        ScrollView {
+            SearchBar(searchText: $searchText)
+            VStack(alignment: .leading) {
+                ForEach(CategorySection.allCases, id: \.self) { section in
+                    let filteredQuizzes = searchResults
+                        .filter { $0.category == section.rawValue.uppercased() }
+                    if !filteredQuizzes.isEmpty {
+                        Section(header: Text(section.rawValue)
+                            .sectionHeaderStyle(section)) {
+                                ForEach(filteredQuizzes, id: \.self) { quiz in
+                                    QuizCellView(quiz: quiz)
+                                        .onTapGesture {
+                                            viewModel.showQuizDetails(quiz: quiz)
+                                        }
+                                }
+                                .cornerRadius(30)
+                            }
+                    }
```

Comment:
Everything that's not related to the drawing of the view should be in the view model.